### ISSUE TITLE
Python lists are not shortened when displayed in console

### ISSUE THREAD

**Issue by david-cortes** [rstudio/rstudio#3143]
When producing some long Python console output in the interactive mode from `reticulate`, the output is not shortened (exception for pandas DFs), but rather printed completely regardless of its length.

**Steps to reproduce:**
* Enter Python interactive mode (e.g. `reticulate::repl_python()`).
* Create some long Python list/dict.
* Access the object in the console output.

**Expected behavior:**
* Should shorten the ouput like in other IDEs.

**Actual behavior:**
* Prints the whole object, which can get very slow and will crash the system with large inputs.

Sample code
```python
import numpy as np
a = np.random.normal(size = 10**7)
a.tolist()
```

**Comment by kevinushey** [comment]
Are you aware of any Python module that we could leverage for pretty-printing and smart truncation here?

**Comment by david-cortes** [comment]
No, no idea, sorry. Also looking at pandas' formatter, it seems their print truncation is just a naive implementation at the class-level of concatenating the first-N and last-N: https://github.com/pandas-dev/pandas/blob/5d0daa0522730a6f999ceaf328f63f03dd62d0b4/pandas/io/formats/format.py
and that's why it works in RStudio too

**Comment by david-cortes** [comment]
Still reproducible, but now the behavior is worse in the sense that (a) it will also hang for a while before it attemtps to print the list, (b) it will freeze the console afterwards.
